### PR TITLE
Support <script type="module">

### PR DESCRIPTION
**Pull request by modulesio**
Fixes [THREE.js examples](https://threejs.org/examples/#webgl_animation_cloth).

Contination of https://github.com/exokitxr/exokit/pull/1246.

### CODE REVIEW

**Comment by modulesio** [comment]
Note that this adds the `node` start requirement of `--experimental-vm-modules`.

File: src/DOM.js [hunk 1]
```diff
@@ -1881,13 +1920,10 @@ class HTMLScriptElement extends HTMLLoadableElement {
     });
   }
 
+>>>>>>> master
```
Comment (RangerMauve) [inline src/DOM.js]: Merge conflicts?

Comment (modulesio) [reply src/DOM.js]: Whoops.

Comment (modulesio) [reply src/DOM.js]: https://github.com/exokitxr/exokit/pull/1255/commits/74b7c1a61cc90e8ec77543cf6e130db98c6bc276

File: src/DOM.js [hunk 2]
```diff
@@ -1805,62 +1805,67 @@ class HTMLScriptElement extends HTMLLoadableElement {
...
 
-          cb();
+          if (this.getAttribute('type') === 'module') {
+            opts.url = url;
+            const script = new vm.SourceTextModule(s, opts);
+            await script.link(async (url, {url: baseUrl}) => {
+              url = _mapUrl(_normalizeUrl(url, baseUrl), this.ownerDocument.defaultView);
+              const s = await _fetch(url);
+              return new vm.SourceTextModule(s, {
+                url: baseUrl,
```
Comment (shawwn) [inline src/DOM.js]: You need to use the parent module's `context` here.

```
await script.link(async (url, {url: baseUrl, context}) => {
  ...
    return new vm.SourceTextModule(s, {
      context,
```

See https://nodejs.org/api/vm.html#vm_class_vm_sourcetextmodule

Comment (modulesio) [reply src/DOM.js]: Would that be different from leaving it out and having the context be the global `window` (as intended)?

**Pull request merged**